app/services/ui_notifications/hosts.rb

```diff
@@ -0,0 +1,23 @@
+module UINotifications
+  module Hosts
+    class Base < UINotifications::Base
+      def deliver!
+        if audience.nil?
+          raise(Foreman::Exception, "Invalid owner for #{subject}, unable to send notifications")
+        end
+        super
+      end
+
+      protected
+
+      def audience
+        case subject.owner_type
+        when 'User'
+          ::Notification::AUDIENCE_USER
+        when 'Usergroup'
+          ::Notification::AUDIENCE_GROUP
+        end
```

Comment:
```
else
  initiator 
end
```

or something like that - I guess you want the notification even if you don't set the owner field